pkg/webui/console/lib/hooks/use-stack-configuration.js

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+
+import {
+  selectAsConfig,
+  selectIsConfig,
+  selectNsConfig,
+  selectJsConfig,
+  selectGsConfig,
+} from '../../../lib/selectors/env'
+import getHostFromUrl from '../../../lib/host-from-url'
+
+const components = {
+  IS: 'is',
+  AS: 'as',
+  NS: 'ns',
+  JS: 'js',
+  GS: 'gs',
+}
+
+/**
+ * React hook to access stack configuration of the lorawan-stackcluster
+ * @returns {Object} helpers - The cluster configuration helpers
+ * @returns {Function} helpers.checkAvailable - Checks availability of a stack component in the cluster
+ * @returns {Function} helpers.checkAllAvailable - Checks availability of all stack components in
+ * `components` in the cluster
+ * @returns {Object} helpers.components - Abbreviations of lorawan-stack components
+ * @returns {Object} helpers.configuration - Configuration of the cluster
+ * @returns {Object} helpers.checkSameHost - Compares the components hostname against provided
+ */
+const useStackConfiguration = () => {
```

Comment:
Do we really need a hook for this? The configuration object is not going to change during runtime. I'd think we could simply use plain functions using our env selectors. It just seems overkill to me, especially since we already have three different ways to pass env info (`@withEnv` decorator, env selectors in connect function, using env selectors directly)